docs/specification.md

```diff
@@ -57,33 +51,83 @@ Compile-time value parameters are captured by the `Config` type (parameterized b
 data Config proposalMetadata = Config
   { cDaoName :: Text
   -- ^ Name of the DAO.
-  , cDaoDescription :: Text
+  , cDaoDescription :: Markdown
   -- ^ Description of the DAO.
-  , cProposalCheck :: Lambda (Natural, Proposal proposalMetadata) Bool
+  , cUnfrozenTokenMetadata :: FA2.TokenMetadata
+  -- ^ FA2 metadata for unfrozen token.
+  , cFrozenTokenMetadata :: FA2.TokenMetadata
+  -- ^ FA2 metadata for frozen token.
+  , cProposalCheck :: forall s
+      . (ProposeParams proposalMetadata) & s
+    :-> Bool & s
   -- ^ A lambda used to verify whether a proposal can be submitted.
   -- It checks 2 things: the proposal itself and the amount of tokens frozen upon submission.
   -- It allows the DAO to reject a proposal by arbitrary logic and captures bond requirements
-  -- (i. e. how much one should "freeze" in order to propose).
-  , cUnfrozenTokenMetadata :: FA2Metadata
-  -- ^ FA2 metadata for unfrozen token.
-  , cFrozenTokenMetadata :: FA2Metadata
-  -- ^ FA2 metadata for frozen token.
-  , cDecisionLambda :: Lambda
-      (Proposal proposalMetadata, Storage proposalMetadata)
-      (List Operation, Storage proposalMetadata)
-  -- ^ The decision lambda is executed based on a successful proposal.
-  , cRejectedProposalReturnValue :: Lambda Natural Natural
+  , cRejectedProposalReturnValue :: forall s
+      . (Proposal proposalMetadata) & s
+    :-> ("slash_amount" :! Natural) & s
```

Comment:
Oh, it's getting more and more technical, and now only developers with Haskell and Lorentz experience can understand what's written there. This specification is supposed to be read not only by developers. Some technical knowledge is of course required, but it should be understood by a person who has some programming experience (in any language) and is familiar with Michelson. Things like `forall s`, `&`, `"slash_amount" :!` and maybe something else are very domain specific and should be avoided. `Lambda Natural Natural` is quite understandable by everyone familiar with Michelson, but the current definition is complicated. `lambda a b` can be converted to `forall s. a : s :-> b : s`, but the latter is less readable and the difference is not essential, so we omit such details.

`Markdown` type is also too technical. What is it? How is it mapped to Michelson? And why markdown, where does it come from? Why don't we permit something else? A plain string is the most generic type there.

Reply:
That makes sense, Ivan. I was not thinking it through, I have updated the syntax to be more general now. I've removed the `Markdown`, use the `Lambda` back, and rename the record field as well. Let me know if this is good enough, Ivan.

Reply:
Looks good, removal of prefixes also makes sense.